content/en/guides/directory-structure/components.md

```diff
@@ -153,46 +153,40 @@ If you have components in nested directories such as:
 ```bash
 components/
   base/
-    Button.vue
+      foo/
+         Button.vue
 ```
 
-The component name will be based on its own filename. Therefore, the component will be:
+The component name will be based on its own path directory and filename. Therefore, the component will be:
 
 ```html
-<button />
+<BaseFooButton />
 ```
 
-We recommend you use the directory name in the filename for clarity:
+However, if you want to use custom prefix, then you can use the `prefix` option in `nuxt.config` to add a prefix to all components in a specific folder.
 
 ```bash
 components/
   base/
-    BaseButton.vue
-```
-
-However, if you want to keep the filename as `Button.vue`, then you can use the `prefix` option in `nuxt.config` to add a prefix to all components in a specific folder.
-
-```bash
-components/
-  base/
-   Button.vue
+      foo/
+         Button.vue
 ```
 
 ```bash{}[nuxt.config.js]
 components: {
   dirs: [
-    '~/components',
+    '~/components', // shortcut to { path: '~/components' }
       {
-        path: '~/components/base/',
-        prefix: 'Base'
+        path: '~/components/base/foo/',
+        prefix: 'awesome'
```

Comment:
Default prefix is (with directory name) is to ensure fs path matches component name. This way we can quickly find component based on it's name. `prefix` is an advanced option and mostly usable when creating component libraries. For this docs section, i would suggest simplify example by registering `~/components/base` to remove `Base` prefix.

Reply:
Ok, that's make sense, i'll fix that